### ISSUE TITLE
Adding functionality to arguments

### ISSUE THREAD

**Issue by kesne** [hayes/giraphql#37]
Ideally, I'd love to be able to wrap arguments somehow. This would make things like validation a lot easier (e.g. applying a schema to arguments directly, as opposed to the current workarounds involving attaching schemas to the field itself).

This would probably also make it possible to have a custom plugin that could define both the schema and the arguments at the same time.

**Comment by kesne** [comment]
Spitballing here, from using this a little more I think my ideal API looks something like this:
--- 

By default there's no new behavior that's added anywhere. This therefore should not be a breaking change. The main new capabilities would be added by plugins.

### Arguments Plugin

Plugins would be able to extend arguments, allowing additional parameters to be added to the arguments calls:

```ts
export interface ArgumentType<T> {
  validator?: (value: T) => boolean
}
```

Because there are multiple types of arguments, and the validators could vary based on that, we probably need some way to support that. We could also support that with a mapped interface. This would allow plugins to easily add arbitrary parameters to specific argument types (this also aligns with how most validation libraries are implemented).

```ts
export interface ArgumentsByType {
  string: { email?: boolean },
  int: { positive?: boolean },
  id: { uuid?: boolean },
}
```

The plugin would then provide a hook for when an argument is resolved:

```ts
wrapArgument(fieldConfig) {
  return (value) => {
    if (fieldConfig.validator && !fieldConfig.validator(value)) {
      throw new Error('Validator returned false');
    }
    return value;
  }
}
```

The benefit of this approach would be that the consumer can merely define argument types, and then get validation applied simply by adding the plugin.

### Consumers

The consumer side for most of this would be seamless, merely adding parameters for arguments that you would like to be validated.

```ts
t.string(() => {
  args: {
    name: t.string({ required: true, minLength: 1, maxLength: 25 })
  },
  resolve: (_root, { name }) => `Hello, ${name}`
});
```

**Comment by hayes** [comment]
Started working on a fix for this in #38 that PR adds utils needed to more easily map input fields.  It doesn't add much along the lines of adding options for specific field methods.  I have not found a good way to have that work well with the impossibility of plugins.  I also have some work in progress on a zod plugin #39

**Comment by hayes** [comment]
validation plugin added in latest release:
Successfully published:
 - @giraphql/converter@2.1.0
 - @giraphql/core@2.1.0
 - @giraphql/plugin-directives@2.0.3
 - @giraphql/plugin-example@2.0.3
 - @giraphql/plugin-mocks@2.0.3
 - @giraphql/plugin-relay@2.1.0
 - @giraphql/plugin-scope-auth@2.0.2
 - @giraphql/plugin-simple-objects@2.0.2
 - @giraphql/plugin-smart-subscriptions@2.0.3
 - @giraphql/plugin-sub-graph@2.0.3
 - @giraphql/plugin-validation@1.1.0
